pkg/audit/reporter.go

```diff
@@ -22,6 +22,13 @@ type Reporter struct {
 	maxRetries int
 }
 
+// RecordAuditsInfo is a struct containing arguments/return values for RecordAudits()
+type RecordAuditsInfo struct {
```

Comment:
Let's also use a separate type for `NodeStats` and add a converter if necessary.

Reply:
curious as to the reason why

Reply:
Separating the layers. protobuf is an implementation detail not the core data-structures of the system. Using protobuf in the underlying layers makes codes somewhat harder to test and read.

When you look at the layers there are:

```
database <- ... <- service <- endpoint <- transport
```

protobuf, grpc etc. are all about transport, not the protocol semantics. By depending directly on protobuf this separation less clearer and hence the whole structure less clearer.